internal/catalog/internal/types/errors.go

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package types
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/consul/proto-public/pbresource"
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+var (
+	errMissing                  = errors.New("missing required field")
+	errEmpty                    = errors.New("cannot be empty")
+	errNotDNSLabel              = errors.New(fmt.Sprintf("value must match regex: %s", dnsLabelRegex))
+	errNotIPAddress             = errors.New("value is not a valid IP address")
+	errUnixSocketMultiport      = errors.New("Unix socket address references more than one port")
+	errInvalidPhysicalPort      = errors.New("port number is outside the range 1 to 65535")
+	errInvalidVirtualPort       = errors.New("port number is outside the range 0 to 65535")
```

Comment:
0 is to indicate there is no virtual port and so it wont be available for tproxy?

We could alternatively use the protobuf well known type IntValue instead of just a primitive int which would allow us to detect unset differently set to 0.

Reply:
Right the two options are:

1. ignore where its being used.
2. Use a google.protobuf.UInt32Value instead of the primitive int. This basically is a message that wraps a single value so that you can detect the pointer to the message being nil to see it as unset.

In either case you need to handle unset so it seemed like since we don't need to allow port 0 it would be easier for users to not have to deal with the wrapper.